### PR TITLE
Enable circuit breaker setting on helm template

### PR DESCRIPTION
**Pull request by danielfablius**
# Description

This PR to expose circuit breaker settings to helm template.

# Context

I want to set Settings.spec.gloo.circuitBreakers.maxConnections etc, but I want to do it via helm.

# Checklist:

- [X] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [X] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [X] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [X] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works

BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/7316

### CODE REVIEW

File: install/helm/gloo/generate/values.go [hunk 1]
```diff
@@ -181,20 +181,21 @@ type KnativeProxyInternal struct {
...
+	Aws                           AwsSettings             `json:"aws,omitempty"`
+	RateLimit                     interface{}             `json:"rateLimit,omitempty" desc:"Partial config for Gloo Edge Enterprise’s rate-limiting service, based on Envoy’s rate-limit service; supports Envoy’s rate-limit service API. (reference here: https://github.com/lyft/ratelimit#configuration) Configure rate-limit descriptors here, which define the limits for requests based on their descriptors. Configure rate-limits (composed of actions, which define how request characteristics get translated into descriptors) on the VirtualHost or its routes."`
+	CircuitBreakers               CircuitBreakersSettings `json:"circuitBreakers,omitempty"`
```
Comment (ashishb-solo) [inline install/helm/gloo/generate/values.go:197]: Could we add a description to this field?

**Review by ashishb-solo — COMMENTED** [review]
Looks good - little to add on my part, but you can fix the codegen issues by running `make -B generated-code` and committing the files that change. Thanks!

**Comment by ashishb-solo** [comment]
Also, just as a friendly FYI, you can also use [kubeResourceOverride](https://docs.solo.io/gloo-edge/latest/reference/helm_chart_values/enterprise_helm_chart_values/#helm-chart-kuberesourceoverrides) to merge additional custom fields that aren't present in our Helm templates into the final charts that are produced. This is a simple way to access fields in the Gloo API that are not explicitly exposed via Helm.

For example, in this case, we could modify the fields provided in this pull request by generating the Helm charts with a few additional flags:

    helm template _output/helm/charts/gloo-*.tgz \
        --set settings.kubeResourceOverride.circuitBreakers.maxConnections=1024 \
        --set settings.kubeResourceOverride.circuitBreakers.maxPendingRequests=1024 \
        --set settings.kubeResourceOverride.circuitBreakers.maxRequests=1024 \
        --set settings.kubeResourceOverride.circuitBreakers.maxRetries=3

which I believe achieves the same output charts as provided by the changes in this pull request.

However, this is simply a suggestion for users who wish to access additional fields in the Gloo API that are not exposed via helm. We certainly prefer exposing these values explicitly in our helm charts.

File: install/test/helm_test.go [hunk 2]
```diff
@@ -3498,6 +3498,17 @@ spec:
 						})
 						testManifest.ExpectUnstructured(settings.GetKind(), settings.GetNamespace(), settings.GetName()).To(BeEquivalentTo(settings))
 					})
+					It("correctly sets the gateway circuitBreakers fields in the settings", func() {
```
[the remark this answers is not in the document]
Comment (danielfablius) [reply install/test/helm_test.go:3501]: for the negative test case, i think this test already cover where the gateway settings is made without the circuit breaker
https://github.com/solo-io/gloo/blob/master/install/test/helm_test.go#L3455
but if you think  its not enough, i will make one

**Comment by ashishb-solo** [comment]
It looks like `go.mod` got emptied - not sure why, but I'm guessing it might have happened during code regeneration. Could you run `go mod tidy` and see if that restores the deleted lines in `go.mod`, and if so, commit those changes? Thanks

**Review by gunnar-solo — APPROVED** [review]

**Review by ashishb-solo — APPROVED** [review]

**Pull request merged**